Clean up process start/stop handling

hlinnaka opened the pull request:
* Poll more frequently when waiting for process start/stop. This speeds up startup and shutdown in tests. We did this already in commit 52ce1c9d53, which reduced the interval to 100 ms, but it was inadvertently increased back to 500 ms in commit d42700280f. Reduce it to 100 ms again, for both start and stop operations.

* Harmonize the start and stop loops, printing the dots and notices the same way in both. I considered extracting the logic to a separate retry-function that takes a closure as argument that does the polling, but as long as we only have two copies, the code duplication isn't that bad.

* Remove newline after "Starting pageserver" and "Starting etcd" messages, so that the progress-indicator dots that are printed once a second are printed on the same line. Before:

    Starting pageserver at '127.0.0.1:64000' in '.neon'
    ...
    pageserver started, pid: 2538937

  After:

    Starting pageserver at '127.0.0.1:64000' in '.neon'...
    pageserver started, pid: 2538937

  The "Starting safekeeper" message already got this right.

SomeoneToIgnore reviewed (commented):
LGTM, but postgres changes seem unrelated.

hlinnaka commented:
> LGTM, but postgres changes seem unrelated.

ah yeah, removed.

> https://github.com/neondatabase/neon/commit/d42700280fffab16327949125d4c09c89de51c84 should've done that too, we can do it here optionally to and update main README.md to reflect the changes, since it shows some console output.

Updated. Good catch.

> Or remove those parts entirely, since it's been multiple times already when we needed to stop and rewrite those parts.

Yeah, it goes out of date easily. I think it's still better to an example there, even if the output might be slightly out of date. The exact details of the output are not that important.
